06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js

```diff
@@ -0,0 +1,184 @@
+      ${book[documentPosition].text}`;
+      bookContents.push(InvertedIndex.tokenize(mergedTitleAndText).split(' '));
+      this.setBookTitles(bookname, book[documentPosition].title);
+    });
+    return bookContents;
+  }
+
+  /**
+   * Creates the index for specified book
+   * @param {string} bookname - Name of book to be indexed
+   * @param {object} book - document words in specified book
+   * @return {Promise.<iDexMapper>} - indexes of specified book
+   */
+  createIndex(bookname, book) {
+    const tokenIndex = {};
+    this.numberOfDocuments[bookname] = [];
+    const bookContents = this.createsArray(bookname, book);
+    return new Promise((resolve) => {
+      bookContents.map((eachdocument, documentPosition) => {
+        const documentPositionToInt = parseInt(documentPosition, 10);
+        this.numberOfDocuments[bookname].push(documentPositionToInt);
+        eachdocument.map((eachWord) => {
+          if (tokenIndex[eachWord]) {
+            if (tokenIndex[eachWord].indexOf(documentPositionToInt) === -1) {
+              tokenIndex[eachWord].push(documentPositionToInt);
+            }
+          } else {
+            tokenIndex[eachWord] = [documentPositionToInt];
+          }
+        });
+      });
+      this.iDexMapper[bookname] = tokenIndex;
+      resolve(this.iDexMapper[bookname]);
+    });
+  }
+
+  /**
+   * Get’s indexes created for particular files
+   * @param {String} bookname - name of book to get its indexes
+   * @return {Object.<iDexMapper>} - indexes of specified bookname
+   */
+  getIndex(bookname) {
+    return this.iDexMapper[bookname];
+  }
+
+/**
+ * Searches through one or more indices for words
+ * @param {String} bookname - name of book to search
+ * @param {String} tokens - words to search
+ * @return {Object.<searchResult>} - words in each book specified
+ */
+  searchIndex(bookname, tokens) {
+    tokens = tokens.split(' ');
+    const allBooks = this.iDexMapper;
+    const bookToSearchName = bookname;
+    const searchResult = [];
+    if (bookToSearchName === 'allBooks') {
+      Object.keys(allBooks).map((book) => {
+        const search = this.getSearchResult(book, tokens);
+        searchResult.push(search);
+      });
+    } else {
+      const search = this.getSearchResult(bookToSearchName, tokens);
+      searchResult.push(search);
+    }
+    return searchResult;
+  }
+
+  /**
+   * Get search result for the specified book
+   * @param {String} bookname - name of book to search
+   * @param {String} tokens - words to search
+   * @return {Object<searchResult>} - words in specified books
+   */
+  getSearchResult(bookname, tokens) {
+    const allBooks = this.iDexMapper;
+    if (!(allBooks[bookname])) {
+      return false;
+    }
+    const searchResult = {};
+    searchResult[bookname] = {};
+    tokens.map((word) => {
+      searchResult[bookname][word] = allBooks[bookname][word] || [];
+    });
+    return searchResult;
+  }
+};
+
+module.exports.InvertedIndex = InvertedIndex;
```

Comment:
Did you get a multiple instances of this error _**Expected to return a value in arrow function**_ on this file?